### PR TITLE
PLAT-864 Set proper comparator for column drop-down

### PR DESCRIPTION
**Pull request by daniel-klose**
Before:
image
After:
image

### CODE REVIEW

File: platform-dom/src/main/java/com/softicar/platform/dom/elements/select/value/simple/display/DomSimpleValueSelectDefaultValueComparator.java [hunk 1]
```diff
@@ -59,22 +60,33 @@ public int compare(V first, V second) {
...
+				return compareNullsFirst(Integer::compareTo, (Integer) first, (Integer) second);
+			} else if (isArgumentsCastableTo(first, second, Float.class)) {
+				return compareNullsFirst(Float::compareTo, (Float) first, (Float) second);
+			} else if (isArgumentsCastableTo(first, second, Double.class)) {
+				return compareNullsFirst(Double::compareTo, (Double) first, (Double) second);
+			} else if (isArgumentsCastableTo(first, second, BigDecimal.class)) {
+				return compareNullsFirst(BigDecimal::compareTo, (BigDecimal) first, (BigDecimal) second);
+			} else if (isArgumentsCastableTo(first, second, Long.class)) {
+				return compareNullsFirst(Long::compareTo, (Long) first, (Long) second);
+			}
```
Comment (daniel-klose) [inline platform-dom/src/main/java/com/softicar/platform/dom/elements/select/value/simple/display/DomSimpleValueSelectDefaultValueComparator.java:87]: I know this is ugly, but every implementation of `Number` has a different `compareTo`, therefore I dont really know how to do this better than simple `else if` blocks

File: platform-dom/src/main/java/com/softicar/platform/dom/elements/select/value/simple/display/DomSimpleValueSelectDefaultValueComparator.java [hunk 2]
```diff
@@ -59,22 +60,33 @@ public int compare(V first, V second) {
...
-					result = Comparator.nullsFirst(comparator).compare((IBasicItem) first, (IBasicItem) second);
+					result = compareNullsFirst(comparator, (IBasicItem) first, (IBasicItem) second);
 				}
 			}
 
 			return result;
 		}
 
 		else {
+			if (isArgumentsCastableTo(first, second, Integer.class)) {
```
Comment (oliver-richers) [inline platform-dom/src/main/java/com/softicar/platform/dom/elements/select/value/simple/display/DomSimpleValueSelectDefaultValueComparator.java:77]: for everything but BigDecimal, you could use Number.doubleValue
```
if (isArgumentsCastableTo(first, second, Number.class)) {
    return compareNullsFirst(Double.compareTo, toNumberDoubleValue(first), toNumberDoubleValue(second));
}
```
and event for BigDecimal, i think Number.doubleValue might suffice, if not then do the BigDecimal check before the Number check.

**Review by oliver-richers — CHANGES_REQUESTED** [review]

File: platform-dom/src/main/java/com/softicar/platform/dom/elements/select/value/simple/display/DomSimpleValueSelectDefaultValueComparator.java [hunk 3]
```diff
@@ -59,22 +61,32 @@ public int compare(V first, V second) {
...
 			return result;
 		}
 
 		else {
+			if (isArgumentsCastableTo(first, second, BigDecimal.class)) {
+				return compareNullsFirst(BigDecimal::compareTo, (BigDecimal) first, (BigDecimal) second);
+			} else if (isArgumentsCastableTo(first, second, Number.class)) {
+				Number firstNumber = (Number) first;
+				Number secondNumber = (Number) second;
+				return compareNullsFirst(
```
Comment (oliver-richers) [inline platform-dom/src/main/java/com/softicar/platform/dom/elements/select/value/simple/display/DomSimpleValueSelectDefaultValueComparator.java:83]: this is not necessary, the test `isArgumentsCastableTo` would have failed, if one of the values was `null`

Comment (oliver-richers) [reply platform-dom/src/main/java/com/softicar/platform/dom/elements/select/value/simple/display/DomSimpleValueSelectDefaultValueComparator.java:83]: ok, that method is strange

**Review by oliver-richers — APPROVED** [review]

**Pull request merged**